docs/federation.md

```diff
@@ -0,0 +1,256 @@
+The Shows DGS defines the `Show` type with fields id, title and releaseYear. 
+Note that the `id` field is marked as the key. 
+This indicates to the gateway that the `id` field will be used for identifying the corresponding Show in the Shows DGS and must be specified for federated types.
+```graphql
+type Query {
+  shows(titleFilter: String): [Show]
+}
+
+type Show @key(fields: "id") {
+  id: ID
+  title: String
+  releaseYear: Int
+}
+```
+
+### Extending a Federated Type
+
+To extend a type you redefine the type in your own schema, using directive `@extends` to instruct that it's a type extension.
+`@key` is required to indicate the field that the gateway will use to identify the original `Show` for a query.
+In this case, the key is the `id` field.
+
+```graphql
+type Show @key(fields: "id") @extends {
+  id: ID @external
+  reviews: [Review]
+}
+
+type Review {
+  starRating: Int
+}
+```
+When redefining a type, only the id field, and the fields you're adding need to be listed.
+Other fields, such as `title` for `Show` are provided by the Shows DGS and do not need to be specified unless you are using it in the schema.
+Federation makes sure the fields provided by all DGSs are combined into a single type for returning the results of a query.
+
+## Implementing a Federated Type
+The very first step to get started is to generate Java types that represent the schema.
+This is configured in `build.gradle` as described in the [manual](./generating-code-from-schema.md).
+When running `./gradlew build` the Java types are generated into the `build/generated` folder, which are then automatically added to the classpath.
+
+### Provide an Entity Fetcher
+Let's go through an example of the following query sent to the gateway:
+```graphql
+query {
+  shows {
+    title
+    reviews {
+      starRating
+    }
+  }
+}
+```
+
+The gateway first fetches the list of all the shows from the Shows DGS containing the title and id fields.
+```graphql
+query {
+  shows {
+    __typename
+    id
+    title
+  }
+}
+
+```
+
+Next, the gateway sends the following `_entities` query to the Reviews DGS using the list of `id`s from the first query:
+```graphql
+query($representations: [_Any!]!) {
+  _entities(representations: $representations) {
+    ... on Show {
+      reviews {
+        starRating
+      }
+    }
+  }
+}  
+```
+
+This query comes with the following variables:
+```json
+{
+  "representations": [    
+    {          
+      "__typename": "Show",
+      "id": 1
+    },
+    ,
+    {
+      "__typename": "Show",
+      "id": 2
+    },
+    {
+      "__typename": "Show",
+      "id": 3
+    },
+    {
+      "__typename": "Show",
+      "id": 4
+    },
+    {
+      "__typename": "Show",
+      "id": 5
+    }
+  ]        
+} 
+```
+
+The Reviews DGS needs to implement an `entity fetcher` to handle this query.
+An entity fetcher is responsible for creating an instance of a `Show` Java instance based on the representation in the `_entities` query above.
```

Comment:
This reads strange imo
> creating an instance of a `Show` Java instance
I would remove the _Java instance_  or change the second `instance` to `reference`.

Reply:
Fixed.